### DISCUSSION TITLE
How do you delete a thread?

### DISCUSSION THREAD

**Discussion by zRyder** [serenity-rs/serenity#2864] [Q&A] [answered]
There appears to be no functions for deleting threads, I see some for creating and editing but nothing for deleting.

**Comment by jamesbt365** [comment]
Does calling delete on the GuildChannel object not delete it?

**Reply by zRyder** [reply]
There doesn't appear to be a way to create an instance of a GuildChannel with something like the thread/message ID for instance. I see something like edit which returns a GuildChannel but it seems silly to have to go through some sort of update function to do that. Especially considering the ChannelID struct has functions for creating and editing threads

**Reply by jamesbt365** [reply]
You can do it directly with the ChannelId too.

**Reply by zRyder** [reply]
So something like ChannelID(ThreadID).delete()…?

**Reply by jamesbt365** [reply, marked as answer]
Yes call delete on the `ChannelId`.

**Reply by zRyder** [reply]
I just got around to trying this, this worked
